ledger/participant-integration-api/src/main/scala/platform/indexer/MeteringAggregator.scala

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer
+
+import com.daml.ledger.offset.Offset
+import com.daml.ledger.participant.state.index.v2.MeteringStore
+import com.daml.ledger.participant.state.index.v2.MeteringStore.ParticipantMetering
+import com.daml.ledger.resources.ResourceOwner
+import com.daml.lf.data.Time.Timestamp
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Metrics
+import com.daml.platform.indexer.MeteringAggregator.{toOffsetDateTime, toTimestamp}
+import com.daml.platform.store.appendonlydao.{DbDispatcher, SqlExecutor}
+import com.daml.platform.store.backend.ParameterStorageBackend.LedgerMeteringEnd
+import com.daml.platform.store.backend.{MeteringStorageWriteBackend, ParameterStorageBackend}
+
+import java.sql.Connection
+import java.time.temporal.ChronoUnit
+import java.time.{OffsetDateTime, ZoneOffset}
+import java.util.{Timer, TimerTask}
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object MeteringAggregator {
+
+  class Owner(
+      meteringStore: MeteringStorageWriteBackend,
+      parameterStore: ParameterStorageBackend,
+      metrics: Metrics,
+      period: FiniteDuration = 6.minutes,
+  ) {
+
+    private[platform] def apply(
+        dispatcher: DbDispatcher
+    )(implicit loggingContext: LoggingContext): ResourceOwner[Unit] = {
+      val aggregator = new MeteringAggregator(meteringStore, parameterStore, metrics)
```

Comment:
having the dispatcher in scope here already, and passing the same in parameters for the two methods on MeteringAggregator is a smell: you probably do not need method passing, it would be enough just to add as an additional constructor parameter to MeteringAggregator

Reply:
Ok